src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -0,0 +1,116 @@
+		
+		final Node s = NodeFactory.createVariable("s");
+		final Node p = NodeFactory.createVariable("p");
+		final Node o = NodeFactory.createVariable("o");
+		final org.apache.jena.graph.Triple testQuery = new TriplePatternImpl(s, p, o).asJenaTriple();
+		final Set<org.apache.jena.graph.Triple> queryResults = new HashSet<>();
+		final Iterator<org.apache.jena.graph.Triple> i = vm.getVocabularyMappingAsGraph().find(testQuery);
+		while(i.hasNext()) {
+			queryResults.add(i.next());
+		}
+		assertEquals(queryResults, jenaTriples);
+	}
+	
+	@Test
+	public void TranslateTriplePatternTest() {
+		final Pair<Set<Triple>,Set<org.apache.jena.graph.Triple>> testData= CreateTestTriples();
+		final Set<Triple> testTriples = testData.object1;
+		final VocabularyMapping vm = new VocabularyMappingImpl(testTriples);
+		
+		final Node s = NodeFactory.createLiteral("s1");
+		final Node p = NodeFactory.createURI(RDF.type.getURI());
+		final Node o = NodeFactory.createLiteral("o1");
+		final TriplePattern testTp = new TriplePatternImpl(s, p, o);
+		final SPARQLGraphPattern translation = vm.translateTriplePattern(testTp);
+		
+		/* Test Union
+		assertTrue(translation instanceof SPARQLUnionPattern);
+		final Set<org.apache.jena.graph.Triple> translatedTriples = new HashSet<>();
+		for(final SPARQLGraphPattern i : ((SPARQLUnionPattern) translation).getSubPatterns()) {
+			assert(i instanceof TriplePattern);
+			translatedTriples.add(((TriplePattern) i).asJenaTriple());
+		}
+		*/
+		
+		//Test Intersection
+		assertTrue(translation instanceof BGPImpl);
+		final Set<org.apache.jena.graph.Triple> translatedTriples = new HashSet<>();
+		for(final TriplePattern i : ((BGPImpl)translation).getTriplePatterns()) {
+			translatedTriples.add(i.asJenaTriple());
+		}
+		
+		assertEquals(testData.object2, translatedTriples);
+	}
+	
+	public Pair<Set<Triple>,Set<org.apache.jena.graph.Triple>> CreateTestTriples(){
+		final Set<Triple> testSet = new HashSet<>();
+		
+		Node s = NodeFactory.createLiteral("s1");
+		Node p = OWL.sameAs.asNode();
+		final Node sRes = NodeFactory.createLiteral("s2");
+		testSet.add(new TripleImpl(s, p, sRes));
+		
+		s = RDF.type.asNode();
+		p = OWL.inverseOf.asNode();
+		final Node pRes = NodeFactory.createLiteral("Not type");
+		testSet.add(new TripleImpl(s, p, pRes));
+		
+		s = NodeFactory.createLiteral("o1");
+		p = OWL.equivalentClass.asNode();
+		final Node o = NodeFactory.createBlankNode();
+		testSet.add(new TripleImpl(s, p, o));
+		
+		s = o;
+		//p = OWL.unionOf.asNode();
+		p = OWL.intersectionOf.asNode();
+		final Node o1Res = NodeFactory.createLiteral("o2");
+		testSet.add(new TripleImpl(s, p, o1Res));
+		
+		final Node o2Res = NodeFactory.createLiteral("o3");
+		testSet.add(new TripleImpl(s, p, o2Res));
+		
+		final Set<org.apache.jena.graph.Triple> expectedRes = new HashSet<>();
+		expectedRes.add(new TriplePatternImpl(o2Res, pRes, sRes).asJenaTriple());
+		expectedRes.add(new TriplePatternImpl(o1Res, pRes, sRes).asJenaTriple());
```

Comment:
Here you should call directly the Jena Triple constructor, instead of creating a `TriplePatternImpl` object just to get its inner Jena Triple
```suggestion
		expectedRes.add(new org.apache.jena.graph.Triple(o2Res, pRes, sRes));
		expectedRes.add(new org.apache.jena.graph.Triple(o1Res, pRes, sRes));
```